extensions/lifecycle/tasks/LifecycleRetriggerRestoreTask.js

```diff
@@ -0,0 +1,50 @@
+'use strict'; // eslint-disable-line
+
+const { LifecycleRequeueTask } = require('./LifecycleRequeueTask');
+
+class LifecycleRetriggerRestoreTask extends LifecycleRequeueTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {LifecycleObjectProcessor} proc - object processor instance
+     */
+     constructor(proc) {
+        super(proc);
+        this.processName = 'restore';
+    }
+
+    updateObjectMD(md, try_, log) {
+        if (this.shouldSkipObject(md, log)) {
+            return false;
+        }
+        md.setUserMetadata({
+            'x-amz-meta-scal-s3-restore-attempt': try_,
+        });
+        md.setOriginOp('s3:ObjectRestore:Retry');
+        return true;
+    }
+
+    shouldSkipObject(md, log) {
+        const isObjectAlreadyRestored = md.getArchive()
+            && md.getArchive().restoreCompletedAt
+            && new Date(md.getArchive().restoreWillExpireAt) >= new Date();
```

Comment:
if object is restored (getArchive && md.getArchive().restoreCompletedAt), but restoreWillExpireAt < date : we should probably not retrigger the restore...

```suggestion
        const isObjectAlreadyRestored = md.getArchive()
            && md.getArchive().restoreCompletedAt;
```